src/lib/optimizer/strategy/chunk_pruning_rule.cpp

```diff
@@ -0,0 +1,100 @@
+#include "chunk_pruning_rule.hpp"
+
+#include <boost/variant.hpp>
+
+#include <algorithm>
+#include <iostream>
+
+#include "all_parameter_variant.hpp"
+#include "constant_mappings.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "optimizer/chunk_statistics.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+std::string ChunkPruningRule::name() const { return "Chunk Pruning Rule"; }
+
+bool ChunkPruningRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) {
+  if (node->type() == LQPNodeType::Predicate && node->child_count() == 1) {
```

Comment:
You can assume a PredicateNode to just have one child (can **can** Assert this is the case if you absolutely want to) no need to clutter this condition with such a check.

Reply:
This would benefit from some good old “Replace Nested Conditional with Guard Clauses”